packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -74,7 +73,7 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
         // recovered address of the user who signed this message. This is how we manage to shim
         // account abstraction even though the user isn't a contract.
         require(
-            transaction.sender() == address(this),
+            transaction.sender() == Lib_ExecutionManagerWrapper.ovmADDRESS(),
```

Comment:
Ok got it, and this is because you are able to mock calls to ovmADDRESS in the wrapper easily.

It does make me sad to reduce efficiency in such a mission critical codepath, but I think that testability here is definitely more important. At this point the higher leverage improvements to make to the `ECDSAContractAccount` are around the require statements / CREATE safety, not small optimizations that harm testability.